src/adr/0040-generalize-event-bus.md

```diff
@@ -0,0 +1,30 @@
+#40. Generalize EventBus
+
+Date: 2020-06-11
+
+## Status
+
+Proposed
+
+## Context
+
+When searching for users by quota ratio if we set the value of the parameters to 0, the search feature is supposed to return newly created users
+who have not receive any email yet at that point. However, this is not the case due to the quotas are currently being initialize only after
+an user receive the first email.
+
+In order to solve this problem, we need to have some kind of event-driven system so when an new user is created successfully, 
+their quota should also be created immediately after. 
```

Comment:
I would mention that "event driven system allows indirection so that the users api don't need a dependency on the quota components".